### PR TITLE
[4.0] web components fixes

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue #17376 .

### Summary of Changes
- Fix the FOUC for ff
- Fix an infinite loop if the alert had no level defined
- remove the extra script for the loader, code is part of core.js (~140 lines)
- Add the non minified versions of the web components, for both the ES5 and ES6 scripts
- Also the non minified and the maps for the poly fills

### Testing Instructions
Apply patch and test that alerts are still good for all the supported browsers

### Documentation Changes Required

### CODE REVIEW

**Comment by dgrammatiko** [comment]
@C-Lodder can you test this one?

**Comment by ghost** [comment]
I have tested this item :red_circle: unsuccessfully on b95f006216c3432a9ff6f114d61a04230b937c37<br /><br />After "Save" Article got:

bildschirmfoto 2017-08-04 um 15 34 47
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17410">issues.joomla.org/tracker/joomla-cms/17410</a>.</sub>

**Comment by dgrammatiko** [comment]
@franz-wohlkoenig why is this unsuccessfully ?

**Comment by ghost** [comment]
@dgt41 should be a green Alert.

**Comment by dgrammatiko** [comment]
@franz-wohlkoenig this is unrelated to what this PR is doing! If you think that this alert should be green (although the type is message not success) please open an issue.

screen shot 2017-08-04 at 17 17 22

**Comment by dgrammatiko** [comment]
To better understand what I am talking about check the title of your image with the next one:
screen shot 2017-08-04 at 17 24 23

File: tests/javascript/validate/spec.js [hunk 1]
```diff
@@ -125,23 +125,29 @@ define(['jquery', 'testsRoot/validate/spec-setup', 'jasmineJquery'], function ($
 		});
 
 		describe('validate method on #validate-numeric-nan', function () {
-			var elNan = document.getElementById('validate-numeric-nan'),
-				res = document.formvalidator.validate(elNan);
+				var elNan = document.getElementById('validate-numeric-nan'),
```
Comment (C-Lodder) [inline tests/javascript/validate/spec.js]: indentation

**Comment by C-Lodder** [comment]
All seems to be working for me. One thing I'd suggest is setting a min width for the alert. The "Article saved" alert is just way too small.

Other than that, the alert needs to be green, not blue, but as you said, that's not related to this PR.

**Comment by dgrammatiko** [comment]
@C-Lodder let's not try to patch alerts any further till we get the decision for the placement of the notifications etc...

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on c4777412a92210d1947a0ed1074386484358ac02<br /><br />As Colour is not related to this PR.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17410">issues.joomla.org/tracker/joomla-cms/17410</a>.</sub>

**Comment by C-Lodder** [comment]
@wilsonge please merge

**Comment by wilsonge** [comment]
@dgt41 does this PR also fix the alert callback thing we talked about on skype yesterday?

**Comment by dgrammatiko** [comment]
@wilsonge not yet

**Pull request merged**